File: 10922 2 the 9s/solution.c
#include <stdio.h>
#include <string.h>

int sumOfDigits(char *array) {
    int result = 0, len = strlen(array);
    for(int i = 0; i < len; ++i)
        result += array[i] - '0';
    return result;
}

int main() {
    char N[1001];
    while(scanf("%s", N) && strcmp(N, "0") != 0) {
        printf("%s is ", N);
        int degree = 0, temp;
        while((temp = sumOfDigits(N)) > 9 && temp % 9 == 0) {
            ++degree;
            sprintf(N, "%d", temp);
        }
        degree += temp == 9;
        if(degree == 0)
            puts("not a multiple of 9.");
        else
            printf("a multiple of 9 and has 9-degree %d.\n", degree);
    }
    return 0;
}

File: 10922 2 the 9s/solution.java
import java.io.*;

public class Main {
	public static void main (String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String N;
		StringBuilder output = new StringBuilder();
		while((N = br.readLine()) != null && N.charAt(0) != '0') {
		    int degree = 0, temp = 0;
		    for(int i = 0; i < N.length(); ++i)
                temp += N.charAt(i) - '0';
            while(temp > 9 && temp % 9 == 0) {
                ++degree;
                temp = sumOfDigits(temp);
            }
            degree += temp == 9 ? 1 : 0;
            if(degree == 0)
                output.append(N).append(" is not a multiple of 9.\n");
            else
                output.append(N).append(" is a multiple of 9 and has 9-degree ").append(degree).append(".\n");
		}
		System.out.print(output);
	}
	
	static int sumOfDigits(int n) {
        int result = 0;
        while(n > 0) {
            result += n % 10;
            n /= 10;
        }
        return result;
    }
}
